### PR TITLE
ICU-20512 ICU4J: just add test of parse with empty curr symbol, code already works as desired

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-20512
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

In ICU4J, just add a test of parsing with empty currency symbol (corresponds to ICU4C test in #2187), code already behaves as desired per ICU-20512

### CODE REVIEW

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java [hunk 1]
```diff
@@ -6966,4 +6966,94 @@ public void Test21556_CurrencyAsDecimal() {
...
+        if (ppos.getIndex() != 1 || num.doubleValue() != 3.0) {
+            errln("parse with empty curr symbol, expect pos 1, value 3.0; get "
+                                + ppos.getIndex() + ", " + num.doubleValue());
+        }
+
+        ppos.setIndex(0);
+        CurrencyAmount currAmt = df.parseCurrency(textToParse, ppos);
+        if (currAmt != null) {
+            errln("parseCurrency with empty curr symbol, expect null but succeeds");
+        }
```
Comment (richgillam) [inline icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java:6990]: This kind of threw me.  Intuitively, it seems like if the currency symbol is the empty string and you parse a string that doesn't have a currency symbol, it should succeed-- the "currency symbol" in the string matches the "currency symbol" in the formatter.  Why is this not the expected behavior?

IIRC, we see this in practice in a couple of locales that have an infix currency symbol, such as the Cape Verdean escudo.  The "currency symbol" is blank, and the actual currency symbol is the currency decimal separator.  In that case, you'd get a parse error if the currency decimal separator wasn't the expected one.  Is this related to that kind of use case?

**Review by macchiati — APPROVED** [review]

**Review by richgillam — APPROVED** [review]
This looks like it does what the ticket and discussion say it should do.  I gather the issue here was that parse() was doing the wrong thing, and this looks like it tests that well, but I'm a little puzzled by the idea that parseCurrency() should fail in this case, even though that's apparently what it has always done.  But I'm coming to this cold, so I'm going to assume you guys all know what you're doing.  But would you mind filling in the background for me?

**Review by sffc — COMMENTED** [review]
I'm surprised that Java passes this test; the C++ and Java behavior are supposed to be identical. It may be worth investigating further why Java works this way but not C++.

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java (hunk 1 above)
Comment (pedberg-icu) [reply icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java:6990]: @richgillam parseCurrency is intended to parse an arbitrary currency, not the specific currency of the locale associated with the formatter. So if there is no currency symbol itcnnot report which currency it parsed. The standard parse is only intended to parse a specific currency, either the one for the locale or some other symbol that has been specifically set (which might be empty).

**Comment by pedberg-icu** [comment]
> I'm surprised that Java passes this test; the C++ and Java behavior are supposed to be identical. It may be worth investigating further why Java works this way but not C++.

I will file a separate ticket for that. Meanwhile this test will catch any regressions in that behavior.

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java (hunk 1 above)
Comment (richgillam) [reply icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java:6990]: Oh, okay.  That makes total sense then.  Thanks for the explanation.

**Comment by pedberg-icu** [comment]
> > I'm surprised that Java passes this test; the C++ and Java behavior are supposed to be identical. It may be worth investigating further why Java works this way but not C++.
> 
> I will file a separate ticket for that. Meanwhile this test will catch any regressions in that behavior.

@sffc : https://unicode-org.atlassian.net/browse/ICU-22161
